doc/contributing/gn-build.md

```diff
@@ -0,0 +1,112 @@
+# GN Build
+
+Similar to GYP, GN is a build system designed for building Chromium. The
+official builds of Node.js are built with GYP, but GN build files are also
+provided as an unofficial alternative build system.
+
+Currently the GN build is used by:
+
+1. Electron for building Node.js together with Chromium.
+2. V8 for testing the integration of Node.js in CI.
+
+## Files of GN
+
+Node.js contains following GN build files:
+
+* `node.gni` - Public GN arguments for configuring the build.
+* `*/BUILD.gn` - This is the file where GN looks for build rules, in Node.js it
+  just includes the `unofficial.gni` file to avoid accidental modifications.
+* `*/unofficial.gni` - The actual build rules for each component.
+
+## Build with GN
+
+Unlike GYP, the GN tool does not include any built-in rules for compiling a
+project, which means projects building with GN must provide their own build
+configurations for things like how to invoke a C++ compiler. Chromium related
+projects like V8 and skia choose to reuse Chromium's build configurations, and
+we are reusing V8's Node.js integration testing repository
+([node-ci](https://chromium.googlesource.com/v8/node-ci/)) for building Node.js.
+
+### 1. Install `depot_tools`
+
+The `depot_tools` is a set of tools used by Chromium related projects for
+checking out code and managing dependencies, and since we are reusing the infra
```

Comment:
It's better to avoid "we" here since it's not certain what that refers to (especially when Node.js maintainers are not responsible for them).

```suggestion
checking out code and managing dependencies, and since this guide is reusing the infra
```